micov/cli.py

```diff
@@ -392,36 +392,106 @@ def _load_db(dbbase, sample_metadata, features_to_keep):
 
 
 @cli.command()
-@click.option('-pos', '--covered-positions', type=click.Path(exists=True),
-              required=True,
-              help='Covered positions calculated from one or more samples')
-@click.option('-o', '--outdir', type=click.Path(exists=True),
-              required=True, help="Output directory. If new, will be created.")
-@click.option('-g', '--genome-id', type=str,
-              required=True, help="Genome ID of the genome of interest")
-@click.option('-l', '--genome-length', type=int,
-              required=True, help="Length of the genome of interest")
-@click.option('-n', '--bin-num', type=int, default=1000,
-              required=False, help="Number of bins")
-def binning(covered_positions, outdir, genome_id, genome_length, bin_num):
+@click.option("--covered-positions", type=click.Path(exists=True),
+    required=True,
+    help="Parquet file containing the covered positions data")
+@click.option("--sample-metadata", type=click.Path(exists=True),
+    required=True,
+    help="A metadata file with the sample metadata")
+@click.option("--features-to-keep", type=click.Path(exists=True),
+    required=False,
+    help="A file with the features to keep. Must have header")
+@click.option("--metadata-variable", type=str, required=True,
+    help="The variable to consider in the sample metadata")
+@click.option("--length", type=click.Path(exists=True), required=True,
+    help="Genome lengths")
+@click.option("--outdir", type=click.Path(exists=False), required=True,
+    help="Output directory for results",)
+@click.option("--bin-num", type=int, default=1000,
+    help="Number of bins (default: 1000)")
+@click.option("--rank", is_flag=True, default=False,
+    help="Enable ranking (default: False)")
+def binning(
+    covered_positions,
+    sample_metadata,
+    features_to_keep,
+    metadata_variable,
+    length,
+    outdir,
+    bin_num,
+    rank
+):
     """Bin genome positions and quantify read and sample hits across bins."""
-    pos = pl.read_csv(covered_positions, separator='\t',
-                      new_columns=BED_COV_SAMPLEID_SCHEMA.columns,
-                      schema_overrides=BED_COV_SAMPLEID_SCHEMA.dtypes_dict,
-                      has_header=False, skip_rows=1).lazy()
-    pos = pos.filter(pl.col(COLUMN_GENOME_ID).is_in([genome_id]))
+    df_md = parse_sample_metadata(sample_metadata).lazy()
+    df_length = parse_genome_lengths(length).lazy()
+    df_pos = pl.scan_parquet(covered_positions)
+
+    df_pos_md = df_pos.join(
+        df_md, on=COLUMN_SAMPLE_ID, how="left"
+    ).join(
+        df_length, on=COLUMN_GENOME_ID, how="left"
+    )
 
-    bin_df, pos_updated = pos_to_bins(pos, genome_length, bin_num)
+    if features_to_keep:
+        features_to_keep = _first_col_as_set(features_to_keep)
+        df_pos_md = df_pos_md.filter(
+            pl.col(COLUMN_GENOME_ID).is_in(features_to_keep))
+
```

Comment:
There is a lot of heavy lifting here, could this be pushed to a module? Would facilitate testing and reuse.